### DISCUSSION TITLE
GraphAR format

### DISCUSSION THREAD

**Discussion by kesavkolla** [apache/incubator-hugegraph#2149] [Q&A]
Is it possible to support [GraphAR](https://github.com/alibaba/GraphAr)  format for storing and reading graph data from disk.

**Comment by imbajin** [comment]
> Is it possible to support [GraphAR](https://github.com/alibaba/GraphAr) format for storing and reading graph data from disk.

Thanks for the introduction.

Yep, we're also interested in this graph **storage format**, however it needs some time to become an industry **standard**(stable), including providing clients in various languages(or WASM way), then we could try to support it

**Reply by sighingnow** [reply]
Hi @imbajin,

GraphAr committer here. Thanks @kesavkolla for introducing. Actually GraphAr has the plan to move to a vendor-neutral venue, such as Apache, and will continue to be developed in an open and transparent manner. Integrating with GraphAr could help to connect with other graph computing systems that already support GraphAr or will support GraphAr soon, e.g., nebular, neo4j, and graphscope. It would be a win-win choice for both the graph computing system and GraphAr itself (see [our hackernews announcement](https://news.ycombinator.com/item?id=35039108)).

> then we could try to support it

May be we could have a meeting or slack channel or something like to discuss the idea. Any suggements/comments on GraphAr (including but not limited to the APIs, SDKs, docs, internal design and community concerns) are highly appreciated!

Providing clients in various language is also in our roadmap. Currently C++ APIs and Java writer (via Spark) are in fairly matural state, and Rust will be supported soon. I would like to discuss about the idea of "WASM way" that I'm wondering if hugegraph currently provides FFI clients via WASM but I failed to found that in the repo. May I know more background about that topic?

Thanks!

**Reply by imbajin** [reply]
Thanks a lot for the invitation，@javeme @zyxxoo @simon824 could take a look on it？

BTW, we don't have ffi integration now, just refer to Kolla's comment

**Reply by sighingnow** [reply]
> BTW, we don't have ffi integration now, just refer to Kolla's comment

I see. Thanks for making it clear.
